pecos/distributed/xmc/xlinear/train.py

```diff
@@ -0,0 +1,211 @@
+        type=str,
+        default="pifa",
+        help="label embedding method for meta-tree",
+    )
+    parser.add_argument(
+        "-sle",
+        "--sub-label-embedding-method",
+        type=str,
+        default="pifa",
+        help="label embedding method for sub-tree",
+    )
+    # Prediction kwargs
+    parser.add_argument(
+        "-k",
+        "--only-topk",
+        type=int,
+        default=20,
+        metavar="INT",
+        help="the default number of top labels used in the prediction",
+    )
+    parser.add_argument(
+        "-b",
+        "--beam-size",
+        type=int,
+        default=10,
+        metavar="INT",
+        help="the default size of beam search used in the prediction",
+    )
+    parser.add_argument(
+        "--max-leaf-size",
+        type=int,
+        default=100,
+        metavar="INT",
+        help="The max size of the leaf nodes of hierarchical 2-means clustering. Multiple values (separated by comma) are supported and will lead to different individual models for ensembling. (default [100])",
+    )
+    parser.add_argument(
+        "-pp",
+        "--post-processor",
+        type=str,
+        choices=PostProcessor.valid_list(),
+        default="l3-hinge",
+        metavar="STR",
+        help="the default post processor used in the prediction",
+    )
+    parser.add_argument(
+        "--seed", type=int, default=0, metavar="INT", help="random seed (default 0)"
+    )
+    parser.add_argument(
+        "--bias", type=float, default=1.0, metavar="VAL", help="bias term (default 1.0)"
+    )
+    parser.add_argument(
+        "--max-iter",
+        type=int,
+        default=20,
+        metavar="INT",
+        help="max iterations for indexer (default 20)",
+    )
+    parser.add_argument(
+        "-n",
+        "--threads",
+        type=int,
+        default=-1,
+        metavar="INT",
+        help="number of threads to use (default -1 to denote all the CPUs)",
+    )
+    parser.add_argument(
+        "-ms",
+        "--main-speed",
+        type=float,
+        default=0.3,
+        metavar="FLOAT",
+        help="main node cpu speed",
+    )
+    parser.add_argument(
+        "-t",
+        "--threshold",
+        type=float,
+        default=0.1,
+        metavar="VAL",
+        help="threshold to sparsify the model weights.",
+    )
+    parser.add_argument(
+        "--verbose-level",
+        type=int,
+        choices=logging_util.log_levels.keys(),
+        default=2,
+        metavar="INT",
+        help=f"the verbose level, {', '.join([str(k) + ' for ' + logging.getLevelName(v) for k, v in logging_util.log_levels.items()])}. Default 2",
+    )
+
+    return parser
+
+
+def do_train(args):
+    """Distributed CPU training and saving XLinear model"""
+
+    # Distributed communicator
+    mpi_comm = MPIComm()
+
+    # Parse args
+    args_dict = vars(args)
+    train_params = DistributedCPUXLinearModel.TrainParams.from_dict(args_dict, recursive=True)
```

Comment:
To be consistent with the single node `pecos.xmc.xlinear.train`, we should support input a json file for params.

Reply:
Yea we should do that, whether to include it into this PR or leave for future improvement is left to debate.